streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -714,19 +720,34 @@ public boolean process(final long wallClockTime) {
             if (recordInfo.queue().size() == maxBufferedSize) {
                 mainConsumer.resume(singleton(partition));
             }
-        } catch (final StreamsException e) {
-            throw e;
+
+            record = null;
+        } catch (final TimeoutException timeoutException) {
+            if (!eosEnabled) {
+                throw timeoutException;
+            } else {
+                record = null;
+                throw new TaskCorruptedException(Collections.singletonMap(id, changelogPartitions()));
```

Comment:
For this case, we don't need the cached record, as we need to reset the task anyway to cleanup potentially "corrupted" state store.

Reply:
Note that we don't trigger `task.timeout.ms` for this case atm. Because we need to restore state what might talk some time, it seems questionable if we should tigger `task.timeout.ms` for this case of not.

Cf. TaskManager#process() that catches `TimeoutException` and trigger the timeout, but does not catch `TaskCorruptedException`

Reply:
I agree, we shouldn't trigger `task.timeout.ms` until it's back to RUNNING right?

Reply:
Not sure if starting the timeout at all makes sense for this case?

Also, starting it when we transit to RUNNING seems to be a non-trivial change.